pkg/controller/cronjob/cronjob_controller.go

```diff
@@ -119,10 +120,92 @@ func (jm *CronJobController) SyncAll() {
 	glog.V(4).Infof("Found %d groups", len(jobsBySj))
 
 	for _, sj := range sjs {
-		SyncOne(sj, jobsBySj[sj.UID], time.Now(), jm.jobControl, jm.sjControl, jm.podControl, jm.recorder)
+		jobsRemaining := CleanupFinishedJobs(sj, jobsBySj[sj.UID], jm.jobControl, jm.recorder)
+		SyncOne(sj, jobsRemaining, time.Now(), jm.jobControl, jm.sjControl, jm.podControl, jm.recorder)
```

Comment:
OK, that's a good point -- this'll also eliminate the logic to maintain the list of jobs remaining, I actually like it better